### PR TITLE
JBTM-2880 don't use printStackTrace in non-test code

### PR DESCRIPTION
**Pull request by mmusgrov**
https://issues.redhat.com/browse/JBTM-2880

CORE !TOMCAT !AS_TESTS !RTS !JACOCO !XTS !QA_JTA !QA_JTS_OPENJDKORB !PERFORMANCE !LRA !DB_TESTS !mysql !db2 !postgres !oracle !JDK17

I accidentally disabled ConnectionImpleTest so the CORE axis failed. I have just pushed [an update](https://github.com/jbosstm/narayana/compare/47def6c650e225ba707d25b390790ca55b1838ec..07ade9aca1ff46357a7420f2471bce5b4cb0a69d) and disabled the unaffected axes.

Don't need to validate against both jdk's, hence !JDK17

I added a comment on the issue to indicate that there is no need to modify test code.

### CODE REVIEW

File: ArjunaJTA/jdbc/tests/classes/com/hp/mwtests/ts/jdbc/basic/ConnectionImpleTest.java [hunk 1]
```diff
@@ -57,9 +57,9 @@ public class ConnectionImpleTest {
     @Before
     public void before() throws Exception {
         MockitoAnnotations.initMocks(this);
-        when(xaDataSource.getXAConnection()).thenReturn(xaConnection);
-        when(xaConnection.getConnection()).thenReturn(connection);
-        when(xaConnection.getXAResource()).thenReturn(xaResource);
+//        when(xaDataSource.getXAConnection()).thenReturn(xaConnection);
+//        when(xaConnection.getConnection()).thenReturn(connection);
+//        when(xaConnection.getXAResource()).thenReturn(xaResource);
```
Comment (mmusgrov) [inline ArjunaJTA/jdbc/tests/classes/com/hp/mwtests/ts/jdbc/basic/ConnectionImpleTest.java:62]: I will remove these after the current CI test run finishes

**Review by marcosgopen — CHANGES_REQUESTED** [review]
Thank you @mmusgrov for your PR. I noticed you haven't updated all classes from 'narayana/ArjunaCore/arjuna/classes/com/arjuna/ats' (only narayana/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/StateManger.java has been updated). e.g. https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java#L2110-L2112

**Comment by marcosgopen** [comment]
I am sorry @mmusgrov, I think you are still missing 3 classes: see https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/log/LogBrowser.java#L190, https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/stats/TxPerfGraph.java#L171 and https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/OTM.java#L354
It might be a 'search' misbehaviour in the IDE (maybe because of some closed projects).
Looking at the code I found also "System.err.println(e);" (e.g. https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/OTM.java#L360) Do you think it might be replaced by a log too?

**Review by marcosgopen — APPROVED** [review]

**Comment by mmusgrov** [comment]
> Thanks @mmusgrov! I approve the PR. Looking at the code I found also some "System.err.println(e);" occurrencies (e.g. https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/OTM.java#L360) Do you think it might be replaced by a log too?

I deliberately skipped OTM.java because it is a Swing app (used for window-based applications) and uses standard out and standard err and users would not necessarily wish to consult a logging framework when running the GUI.

**Comment by marcosgopen** [comment]
> > Thanks @mmusgrov! I approve the PR. Looking at the code I found also some "System.err.println(e);" occurrencies (e.g. https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/OTM.java#L360) Do you think it might be replaced by a log too?
> 
> I deliberately skipped OTM.java because it is a Swing app (used for window-based applications) and uses standard out and standard err and users would not necessarily wish to consult a logging framework when running the GUI.

Ok, I see. Thanks for the clarification.

File: ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/logging/arjunaI18NLogger.java [hunk 2]
```diff
@@ -1621,6 +1621,26 @@ public void warn_objectstore_JDBCImple_over_max_image_size(int imageSize,
...
+	@LogMessage(level = WARN)
+	void warn_LogStore_1(@Cause() Throwable arg0);
+
+	@Message(id = 12411, value = "Unable to locate object store directory", format = MESSAGE_FORMAT)
+	@LogMessage(level = WARN)
+	void warn_objectstore_directory_location(@Cause() Throwable arg0);
+
+	@Message(id = 12412, value = "Unexpected exception:", format = MESSAGE_FORMAT)
+	@LogMessage(level = WARN)
+	void unexpected_excepton(@Cause() Throwable arg0);
```
Comment (graben) [inline ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/logging/arjunaI18NLogger.java:1642]: Hi @mmusgrov, looks like a little typo in unexpected_except(i)on. :)

**Pull request merged**